### PR TITLE
Feature/NYCCHKBK-10312: PHP Unit test cases for Citywide Contracts URL Service

### PR DESCRIPTION
**Pull request by nranevj**

### CODE REVIEW

File: source/webapp/sites/all/modules/custom/checkbook_widget_redesign/checkbook_business_layer/checkbook_services/budget/BudgetWidgetService.php [hunk 1]
```diff
@@ -102,6 +102,6 @@ public function adjustParameters($parameters, $urlPath) {
     }
 
     public function getWidgetFooterUrl($parameters) {
-        return BudgetUrlService::getFooterUrl($parameters,$this->getLegacyNodeId());
```
Comment (nranevj) [inline source/webapp/sites/all/modules/custom/checkbook_widget_redesign/checkbook_business_layer/checkbook_services/budget/BudgetWidgetService.php:105]: $parameters was removed from function definition, as it is not used anywhere inside the function. This is the only place that this function is called. Hence removed $parameters.

File: source/webapp/sites/all/modules/custom/checkbook_project/includes/checkbook_project.inc [hunk 2]
```diff
@@ -1969,7 +1969,7 @@ function _checkbook_append_url_params($source = null, $overidden_params = array(
     if (($current_url[3] == 'contract' && ($current_url[4] == 'search' || $current_url[4] == 'all') && $current_url[5] == 'transactions')) {
       $advanced_search = true;
     }
-    if (!$advanced_search) {
+    if (isset($advanced_search) && !$advanced_search) {
```
Comment (nranevj) [inline source/webapp/sites/all/modules/custom/checkbook_project/includes/checkbook_project.inc:1972]: Gives undefined variable error, and adding this check won't impact the current logic/program flow.

File: source/webapp/sites/all/modules/custom/checkbook_widget_redesign/checkbook_business_layer/checkbook_services/budget/BudgetWidgetService.php (hunk 1 above)
Comment (sgade-reisys) [reply source/webapp/sites/all/modules/custom/checkbook_widget_redesign/checkbook_business_layer/checkbook_services/budget/BudgetWidgetService.php:105]: @nranevj, can  you please make sure the is change does not impact widget footer url on the UI also?

Comment (nranevj) [reply source/webapp/sites/all/modules/custom/checkbook_widget_redesign/checkbook_business_layer/checkbook_services/budget/BudgetWidgetService.php:105]: @sgade-reisys this change does not impact the widget footer URL on the UI. Have confirmed the same by comparing local URLs with production ones.

**Review by sgade-reisys — APPROVED** [review]

**Pull request merged**